### PR TITLE
Fix empty multipart body for client

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

**Comment by e5l** [comment]
Hey @Agcom, could you take a look?

**Comment by Agcom** [comment]
@e5l 
I mentioned (in the issue) that the problem is not related to any particular client.
You can also use curl to reproduce the same behavior.
```
curl -X POST localhost:8080 --header "content-type:multipart/formdata; boundary=-----"
```
The changes are just pointing to the Ktor client core. I don't think this solves the issue.

**Comment by e5l** [comment]
It solves the multipart body in client at least. Let's fix it step by step.

File: ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/TestServer.kt [hunk 1]
```diff
@@ -22,7 +21,7 @@ internal fun startServer(): Closeable {
 
     val proxyServer = TestTcpServer(HTTP_PROXY_PORT, ::proxyHandler)
 
-    val server = embeddedServer(Netty, DEFAULT_PORT) {
+    val server = embeddedServer(CIO, DEFAULT_PORT) {
```
Comment (Ololoshechkin) [inline ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/TestServer.kt:24]: Why do you change to CIO everywhere in tests? How is it related to the fix of empty multipart body?

Comment (e5l) [reply ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/TestServer.kt:24]: This test strikes only with `CIO`. `Netty` ignores wrong prepared request

File: ktor-server/ktor-server-test-host/build.gradle [hunk 2]
```diff
@@ -26,8 +26,6 @@ kotlin.sourceSets.jvmMain.dependencies {
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-debug:$coroutines_version"
 }
 
-if (!project.ext.ideaActive) {
```
Comment (Ololoshechkin) [inline ktor-server/ktor-server-test-host/build.gradle:29]: Why do we need ideaActive at all?

Comment (Ololoshechkin) [reply ktor-server/ktor-server-test-host/build.gradle:29]: or why did we need*

Comment (e5l) [reply ktor-server/ktor-server-test-host/build.gradle:29]: It was a bug in old kotlin plugin, the didn't support custom Gradle configurations

**Pull request merged**